integration_tests/tests/consistency/consistency_columns.sql

```diff
@@ -0,0 +1,52 @@
+{{ config(
+    tags="fivetran_validations",
+    enabled=var('fivetran_validation_tests_enabled', false)
+) }}
+
+/* This test is to make sure the final columns produced are the same between versions.
+Only one test is needed since it will fetch all tables and all columns in each schema.
+!!! THIS TEST IS WRITTEN FOR BIGQUERY!!! */
+{% if target.type == 'bigquery' %}
+with prod as (
+    select
+        table_name,
+        column_name,
+        /* Need the case since the prod version is uncasted, and bigquery automatically produces a value of "bignumeric" or "float"
+        while the dev is casted as dbt.type_numeric() and produces a value of "numeric" */
+        case when lower(data_type) like '%numeric%' or lower(data_type) like '%float%' then 'numeric'
+            else data_type 
+            end as data_type
+    from {{ target.schema }}_salesforce_prod.INFORMATION_SCHEMA.COLUMNS
+),
+
+dev as (
+    select
+        table_name,
+        column_name,
+        case when lower(data_type) like '%numeric%' or lower(data_type) like '%float%' then 'numeric'
+            else data_type 
+            end as data_type
+    from {{ target.schema }}_salesforce_dev.INFORMATION_SCHEMA.COLUMNS
+),
+
+final as (
+    -- test will fail if any rows from prod are not found in dev
+    (select * from prod
+    except distinct
+    select * from dev)
+
+    union all -- union since we only care if rows are produced
+
+    -- test will fail if any rows from dev are not found in prod
+    (select * from dev
+    except distinct
+    select * from prod)
+)
```

Comment:
This test is great! However, if there is a failure it is really difficult to decipher. Which record is from the prod/dev test and which is from the dev/prod test. Would there be a way to make this easier to understand when there is a failure?

Not required in this update, but I know we are reusing this a lot and would like to think about this for a future enhancement.

Reply:
Good point. I found a way and updated!